src/System.Private.ServiceModel/tools/test/Bridge/WcfTestBridgeCommon/CertificateManager.cs

```diff
@@ -173,18 +194,48 @@ public static X509Certificate2 CreateAndInstallLocalMachineCertificates(Certific
 
                 // always create a certificate locally for the current machine's fully qualified domain name, 
                 // hostname, and "localhost". 
-                var hostCert = certificateGenerator.CreateMachineCertificate(fqdn, hostname, "localhost").Certificate;
+                CertificateCreationSettings certificateCreationSettings = new CertificateCreationSettings() { Subjects = new string[] { fqdn, hostname, "localhost" } };
+                var hostCert = certificateGenerator.CreateMachineCertificate(certificateCreationSettings).Certificate;
 
                 // Since s_myCertificates keys by subject name, we won't install a cert for the same subject twice
                 // only the first-created cert will win
                 InstallCertificateToRootStore(rootCertificate);
-                InstallCertificateToMyStore(hostCert);
+                InstallCertificateToMyStore(hostCert, certificateCreationSettings.IsValidCert);
                 s_localCertificate = hostCert;
             }
 
             return s_localCertificate;
         }
 
+        public static X509Certificate2 CreateAndInstallNonDefaultMachineCertificates(CertificateGenerator certificateGenerator, CertificateCreationSettings certificateCreationSettings, string resourceAddress)
+        {
+            if (certificateCreationSettings == null)
+            {
+                throw new ArgumentException("certificateCreationSettings cannot be null as we are creating a non default certificate");
+            }
+
+            if (certificateGenerator == null)
+            {
+                throw new ArgumentNullException("certificateGenerator");
+            }
+
+            lock (s_certificateLock)
+            {
+                Trace.WriteLine("[CertificateManager] Installing Non default Machine certificates to machine store.");
+
+                // At this point, we know we haven't generated the certs yet, or the operation is completing on another thread
+                // Certificate generation is time-consuming, so we want to make sure that we don't unnecessarily generate a cert
```

Comment:
NIT: this comment is not necessarily correct here :) it was true before based on the flow of the code, but it might not be here.